test/src/unit-cppapi-deletes.cc

```diff
@@ -397,6 +444,59 @@ bool DeletesFx::is_array(const std::string& array_name) {
   return vfs_.is_dir(array_name);
 }
 
+/* List the given path and store the counts of each file type. */
+struct CommitsDirectory {
```

Comment:
Let's make this a class and the members private and have accessors. I also think we should put it in common code.

Reply:
We could also search and replace other parts of the code doing this and have them use this. For example, search for void check_num_commits_files(